spec/system/bus_stops/managing_bus_stops_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+
+describe 'managing stops as an admin' do
+  before :each do
+    admin = create :user, :admin
+    @bus_stop = create :bus_stop
+    
+    when_current_user_is admin
+    visit manage_bus_stops_url
+  end
+  context 'delete button' do
+    it 'deletes the specific bus stop' do
+      expect(page).to have_selector 'table.manage tbody tr', count: 1
+      within 'tr', text: @bus_stop.name do
+        click_button 'Delete'
+      end
+      page.driver.browser.switch_to.alert.accept
```

Comment:
Could you walk through what this does? Haven't seen this before in our system tests.

Reply:
It accepts the js alert box

Reply:
Ah, okay. I suppose I could have guessed that.

If we think we might reuse this, it could be worth putting this in a helper method. That would also give us an opportunity to ascribe a descriptive name to the code.

It's a little unusual to see six short methods chained like that, is why it caught my attention.

Reply:
Alright, sounds good